### PR TITLE
Add types for fast-astar:1.0

### PR DESCRIPTION
**Pull request by shezard**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/fast-astar/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    set(coords: [number, number], value: 'value', passable: 0 | 1): void;
+}
+
+export class Astar {
+    constructor(grid: Grid);
+    search(
+        originCoords: [number, number],
+        destinationCoords: [number, number],
+        options: { rightAngle: boolean }
+    ): undefined | Array<[number, number]>;
```
Comment (peterblazejewicz) [inline types/fast-astar/index.d.ts:17]: here and in other places, IMO would be nice to have named tuples (already supported on DT without any TS requirements), probably as `[x: number, y:number]` (or whatever are proper names for these tuple's values]

Comment (shezard) [reply types/fast-astar/index.d.ts]: You are right, i've done the modification !

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@shezard thanks!

**Comment by shezard** [comment]
Ready to merge

**Pull request merged**